astropy/time/time_helper/function_helpers.py

```diff
@@ -0,0 +1,46 @@
+"""
+Helpers for overriding numpy functions in
+`~astropy.time.Time.__array_function__`.
+"""
+import numpy as np
+
+from astropy.units.quantity_helper.function_helpers import FunctionAssigner
+
+# TODO: Fill this in with functions that don't make sense for times
+UNSUPPORTED_FUNCTIONS = {}
+# Functions that take two Time arguments and return a single Time
+TWOARG_HELPERS = {}
+
+function_helper = FunctionAssigner(TWOARG_HELPERS)
+
+
+@function_helper(helps={np.linspace})
+def two_arg(t1, t2, *args):
+    """
+    Parameters
+    ----------
+    t1, t2 :
+        Time-like arguments given to numpy function. At least one of these is
+        an `astropy.time.Time`, and the other should be castable to `Time`.
+    *args :
+        Additional arguments given to the numpy function.
+
+    Returns
+    -------
+    t : astropy.time.Time
+        One of the times, used to keep track of the time scale, location, and
+        format.
+    args : tuple
+        args to pass to the numpy function. The first two are ``t1.jd, t2.jd``.
+    """
+    from astropy.time import Time
+    if isinstance(t1, Time):
+        t = t1
+        t2 = Time(t2)
+        t2 = getattr(t2, t.scale)
+    else:
+        t = t2
+        t1 = Time(t1)
+        t1 = getattr(t1, t.scale)
+
+    return t, (t1.jd, t2.jd, *args)
```

Comment:
I think for `np.linspace`, I would not go through `jd`, which will lose some precision, but rather actually create the result here, via something like
```
offsets = np.linspace(np.zeros(t1.shape), np.ones(t2.shape, *args, **kwargs)
return t1 + (t2-t1)*offsets
```
(i.e., in terms of what is done in `quantity_helpers/function_helpers`, I think this function makes most sense as a dispatched function).

Reply:
p.s. Strictly speaking, the above is not full precision either. Arguably better would be
```
t1 + ((t2 - t1) * np.arange(n)) / (n if endpoint else n+1)
```

Reply:
I'm hesitant with the second option as it's re-implementing the `endpoint` logic in numpy.  Why is it fuller precision?

Reply:
It's higher precision because the multiplication and division will be done in using both jd1 and jd2 (effectively in double double precision), and one is multiplying and dividing by integers here (i.e., `t1 + (t2-t1) * (np.arange(n)/n)` would not have the same benefit).

But I think the first suggestion is probably be OK too, since `t2-t1` will generally be much smaller than `t1`.